### PR TITLE
SPA

### PR DESCRIPTION
**Pull request by natemoo-re**
Just here for the preview link

### CODE REVIEW

File: src/layouts/MainLayout.astro [hunk 1]
```diff
@@ -55,5 +55,27 @@ const { ...head } = Astro.props
...
+						if (cache.has(href)) continue
+
+						promises.push(
+							fetch(new URL(href, window.location.toString())).then((res) => {
+								cache.add(href)
+							}),
+						)
+					}
+					return Promise.race([
+						new Promise((resolve) => setTimeout(resolve, 100)),
```
Comment (jasikpark) [inline src/layouts/MainLayout.astro:74]: what is all of this??? the animation is so nice tho

Comment (natemoo-re) [reply src/layouts/MainLayout.astro]: This is a hacky way to preload the incoming CSS, if there is any

Comment (natemoo-re) [reply src/layouts/MainLayout.astro]: Removed! Not needed.

File: src/layouts/MainLayout.astro [hunk 2]
```diff
@@ -62,25 +62,7 @@ const { ...head } = Astro.props
...
-							}),
-						)
-					}
-					return Promise.race([
-						new Promise((resolve) => setTimeout(resolve, 200)),
-						Promise.all(promises),
-					])
-				},
-			})
+			listen()
```
Comment (jasikpark) [inline src/layouts/MainLayout.astro:65]: hey, listen

**Comment by tony-sull** [comment]
Closing this out for now but I'm *very* interested to reopen this PR for a few more preview builds when I have a chance to really put it through its paces!

I want to really hammer on the performance a bit since this would add a SPA-mode with DOM diffing, very curious to see how it stacks up!

**Pull request closed without merging**